ENIGMAsystem/SHELL/Universal_System/Extensions/Asynchronous/ASYNCdialog.h

```diff
@@ -21,6 +21,20 @@
 #include <string>
 using std::string;
 
+namespace enigma {
+
+struct MessageData {
+  unsigned id;
+  string text1;
+  string text2;
+  string text3;
```

Comment:
This is really weird... is this an artifact from GML?

Reply:
This is an artifact from GMSv1.4. They added async events for things like `sprite_add_async` and `show_message_async` instead of giving the user proper thread functions. So I made the extension by just using our user thread functions to call the internal dialog functions with the appropriate data.